vllm/v1/core/sched/scheduler.py

```diff
@@ -1290,4 +1290,9 @@ def _update_from_kv_xfer_finished(self,
             self.finished_recving_kv_req_ids.add(req_id)
         for req_id in (kv_connector_output.finished_sending or ()):
             logger.debug("Finished sending KV transfer for request %s", req_id)
-            self._free_blocks(self.requests[req_id])
+            if req_id not in self.requests:
+                logger.warning(
+                    "Got finished sending KV transfer for request %s,"
+                    "but the request is already freed.", req_id)
```

Comment:
How would you unit test this scenario?

In #25067 the case I tested was an abort after the prefill request had finished, but @NickLucche rightly asked (AIUI):

> If P is done with the request (finished, capped length), then how is the request being aborted in P ?

Whatever the scenario is ... if it is supposed to happen, we shouldn't have a warning that is not actionable by the user. But a "why this is supposed to happen" comment would be important for maintainability. Something similar to https://github.com/vllm-project/vllm/blob/8616300ae2a00b48aee2f02220e9e007f76ce77c/vllm/v1/core/sched/scheduler.py#L888-L893